src/code/z_camera.c

```diff
@@ -1,3070 +1,6778 @@
+        Camera_Vec3fVecSphGeoAdd(eyeNext, at, &sp7C);
+
+        *eye = *eyeNext;
+        anim->animTimer--;
+
+        if (camera->globalCtx->envCtx.skyDisabled == 0) {
+            Camera_BGCheck(camera, at, eye);
+        } else {
+            func_80044340(camera, at, eye);
+        }
+    } else {
+        sp58 = Math_Sins(-sp60.rot.x);
+        temp_f0_3 = Math_Coss(-sp60.rot.x);
+        sp98.x = subj3->atOffset.x;
+        sp98.y = (subj3->atOffset.y * temp_f0_3) - (subj3->atOffset.z * sp58);
+        sp98.z = (subj3->atOffset.y * sp58) + (subj3->atOffset.z * temp_f0_3);
+        sp58 = Math_Sins(BINANG_ROT180(sp60.rot.y));
+        temp_f0_3 = Math_Coss(BINANG_ROT180(sp60.rot.y));
+        subj3->atOffset.x = (sp98.z * sp58) + (sp98.x * temp_f0_3);
+        subj3->atOffset.y = sp98.y;
+        subj3->atOffset.z = (sp98.z * temp_f0_3) - (sp98.x * sp58);
+        at->x = subj3->atOffset.x + sp60.pos.x;
+        at->y = subj3->atOffset.y + sp60.pos.y;
+        at->z = subj3->atOffset.z + sp60.pos.z;
+        sp7C.r = subj3->eyeNextDist;
+        sp7C.yaw = BINANG_ROT180(sp60.rot.y);
+        sp7C.pitch = sp60.rot.x;
+        Camera_Vec3fVecSphGeoAdd(eyeNext, at, &sp7C);
+        sp7C.r = subj3->eyeDist;
+        Camera_Vec3fVecSphGeoAdd(eye, at, &sp7C);
+    }
+
+    camera->posOffset.x = camera->at.x - playerPosRot->pos.x;
+    camera->posOffset.y = camera->at.y - playerPosRot->pos.y;
+    camera->posOffset.z = camera->at.z - playerPosRot->pos.z;
+    camera->fov = Camera_LERPCeilF(subj3->fovTarget, camera->fov, 0.25f, 1.0f);
+    camera->roll = 0;
+    camera->atLERPStepScale = 0.0f;
+    return 1;
+}
+
+s32 Camera_Subj4(Camera* camera) {
+    Vec3f* eye = &camera->eye;
+    Vec3f* eyeNext = &camera->eyeNext;
+    Vec3f* at = &camera->at;
+    u16 spAA;
+    Vec3s* spA4;
+    Vec3f sp98;
+    Vec3f sp8C;
+    f32 sp88;
+    s16 pad2;
+    f32 temp_f16;
+    PosRot sp6C;
+    VecSph sp64;
+    VecSph sp5C;
+    s16 temp_a0;
+    f32 tx;
+    Player* player;
+    PosRot* playerPosRot = &camera->playerPosRot;
+    Subj4* subj4 = (Subj4*)camera->paramData;
+    Subj4Anim* anim = &subj4->anim;
+
+    if (RELOAD_PARAMS) {
+        CameraModeValue* values = sCameraSettings[camera->setting].cameraModes[camera->mode].values;
+        subj4->interfaceFlags = NEXTSETTING;
+    }
+
+    if (R_RELOAD_CAM_PARAMS) {
+        Camera_CopyPREGToModeValues(camera);
+    }
+
+    if (camera->globalCtx->view.unk_124 == 0) {
+        camera->globalCtx->view.unk_124 = (camera->thisIdx | 0x50);
+        anim->unk_24 = camera->xzSpeed;
+        return true;
+    }
+
+    func_8002EF44(&sp6C, &camera->player->actor);
+
+    OLib_Vec3fDiffToVecSphGeo(&sp5C, at, eye);
+    sCameraInterfaceFlags = subj4->interfaceFlags;
+    if (camera->animState == 0) {
+        spA4 = Camera_GetCamBgDataUnderPlayer(camera, &spAA);
+        Camera_Vec3sToVec3f(&anim->unk_00.a, &spA4[1]);
+        Camera_Vec3sToVec3f(&sp98, &spA4[spAA - 2]);
+
+        sp64.r =
```

Comment:
can these fields be referenced by name now instead of offset?

Reply:
.. why did github keep highlighting this code instead lol
Idr what this was for now, but theres some function comment that says "camera + 0xoffset"

Reply:
/**
 * Camera follows points specified at camera + 0x124 and camera + 0x128
 * until all keyFrames have been exhausted.
 */
s32 Camera_Demo1(Camera* camera) {